packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -117,6 +140,14 @@ function TabsComponent(props: Props): JSX.Element {
       blockElementScrollPositionUntilNextRender(newTab);
       setSelectedValue(newTabValue);
 
+      if (queryString) {
+        const searchKey =
+          typeof queryString === 'string' ? queryString : groupId || '';
+        const searchParams = new URLSearchParams(location.search);
+        searchParams.set(searchKey, newTabValue);
+        history.push({...location, search: searchParams.toString()});
```

Comment:
I would rather use replace than push. push creates new history entries ie you'll move back to previous tab when navigating. Maybe we could make this configurable later but replace looks like a better default for this specific UX. 

I can't test it because no more dogfood way => we need some, please add some docs about that new attribute, that should make it easier to review/test the PR through the deploy preview

Reply:
`replace` used instead of `push`

Also added docs which should be enough for dogfooding, but let me know if you would like me to add more.